### DISCUSSION TITLE
It's not SQLite replacement if not `no_std`

### DISCUSSION THREAD

**Discussion by michalfita** [tursodatabase/turso#441] [General]
One of key features of original SQLite is that in can be built for bare metal system and used without OS in place on microcontrollers. To achieve the same, the core of Limbo would have to be `no_std` with ability to plug in a storage backend that's non-Linux thing. But I don't see `#![no_std]` clause in core's `lib.rs`.

Is this no-feature for Limbo being oriented for cloud solutions?

**Comment by penberg** [comment]
I actually don't mind supporting `no_std`! Just not something that has been on my radar. I/O is already pluggble so we're not dependent on Linux. If someone wants to work on this, would love to see it happen!

**Reply by michalfita** [reply]
The hard part: all your dependencies have to support `no_std`.

**Reply by penberg** [reply]
@michalfita Sure, but I would actually anyway prefer that. Let me convert this into an issue.
